python/mxnet/gluon/block.py

```diff
@@ -841,6 +842,140 @@ def hybrid_forward(self, F, x, *args, **kwargs):
         # pylint: disable= invalid-name
         raise NotImplementedError
 
+
+class BlockList(Block):
+    r"""Holds sub-blocks in a list.
+
+    BlockList can be indexed like a regular Python list, but blocks it
+    contains are properly registered, and will be visible by all Block methods::
```

Comment:
what does it mean to be "properly" registered? Can a block be improperly registered?